src/storage/mvcc/reader/reader.rs

```diff
@@ -166,6 +169,28 @@ impl<S: EngineSnapshot> MvccReader<S> {
             write_cursor: None,
             scan_mode,
             current_key: None,
+            hint_min_version: None,
+            fill_cache: !ctx.get_not_fill_cache(),
+            term: ctx.get_term(),
+            version: ctx.get_region_epoch().get_version(),
+        }
+    }
+
+    pub fn new_min_version_hint_with_ctx(
```

Comment:
how about passing `hint_min_version` in `scan_writes`, you can add a parameter for `create_write_cursor`.